### PR TITLE
refact: remove snaps

### PR DESCRIPTION
**Pull request by jeevithakannan2**
## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Documentation update
- [x] Refactoring
- [ ] Hotfix
- [ ] Security patch
- [ ] UI/UX improvement

## Description
- Check if snap is installed before trying to remove snap.
- Merge dnf and zypper packagers as they both have the same commands.

error

no-error

## Testing
- Works as expected

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

File: core/tabs/system-setup/remove-snaps.sh [hunk 1]
```diff
@@ -3,27 +3,29 @@
...
+        case "$PACKAGER" in
+            pacman)
+                "$ESCALATION_TOOL" "$PACKAGER" -Rns snapd
+                ;;
+            apt-get|nala)
+                "$ESCALATION_TOOL" "$PACKAGER" autoremove --purge snapd
+                if [ "$ID" = ubuntu ]; then
+                    "$ESCALATION_TOOL" apt-mark hold snapd
+                fi
+                ;;
```
Comment (nnyyxxxx) [inline core/tabs/system-setup/remove-snaps.sh:16]:
```suggestion
    if command_exists snap; then
        case "$PACKAGER" in
            apt-get|nala)
                "$ESCALATION_TOOL" "$PACKAGER" autoremove --purge snapd
                if [ "$ID" = ubuntu ]; then
                    "$ESCALATION_TOOL" apt-mark hold snapd
                fi
                ;;
```

Snapd is not a valid package on arch linux
image

**Review by nnyyxxxx — CHANGES_REQUESTED** [review]

File: core/tabs/system-setup/remove-snaps.sh [hunk 2]
```diff
@@ -3,27 +3,29 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" remove snapd
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
+                exit 1
+                ;;
+        esac
+        printf "%b\n" "${GREEN}Successfully removed snaps...${RC}"
+    else
+        printf "%b\n" "${GREEN}Snap is not installed...${RC}"
```
Comment (nnyyxxxx) [inline core/tabs/system-setup/remove-snaps.sh:27]:
```suggestion
        printf "%b\n" "${GREEN}Snapd is not installed...${RC}"
```

File: core/tabs/system-setup/remove-snaps.sh (hunk 1 above)
Comment (jeevithakannan2) [reply core/tabs/system-setup/remove-snaps.sh:16]: It's a AUR package. AUR packages can be uninstalled using pacman

Comment (nnyyxxxx) [reply core/tabs/system-setup/remove-snaps.sh:16]: ah i just noticed that it was in the aur

File: core/tabs/system-setup/remove-snaps.sh [hunk 3]
```diff
@@ -3,27 +3,29 @@
...
+                "$ESCALATION_TOOL" "$PACKAGER" remove snapd
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
+                exit 1
+                ;;
+        esac
+        printf "%b\n" "${GREEN}Successfully removed snaps...${RC}"
+    else
+        printf "%b\n" "${GREEN}Snapd is not installed...${RC}"
```
Comment (adamperkowski) [inline core/tabs/system-setup/remove-snaps.sh:27]:
```suggestion
        printf "%b\n" "${GREEN}Snapd is not installed.${RC}"
```

File: core/tabs/system-setup/remove-snaps.sh [hunk 4]
```diff
@@ -3,27 +3,29 @@
...
+                ;;
+            dnf|zypper)
+                "$ESCALATION_TOOL" "$PACKAGER" remove snapd
+                ;;
+            *)
+                printf "%b\n" "${RED}Unsupported package manager: ""$PACKAGER""${RC}"
+                exit 1
+                ;;
+        esac
+        printf "%b\n" "${GREEN}Successfully removed snaps...${RC}"
```
Comment (adamperkowski) [inline core/tabs/system-setup/remove-snaps.sh:25]:
```suggestion
        printf "%b\n" "${GREEN}Successfully removed snaps.${RC}"
```

**Review by adamperkowski — COMMENTED** [review]
NIT (again)

**Comment by jeevithakannan2** [comment]
> NIT (again)

Unless the end user thinks there is an error occurred when uninstalling snap when the snap is not even installed

**Review by ChrisTitusTech — APPROVED** [review]

**Pull request merged**